controllers/machinehealthcheck_targets.go

```diff
@@ -135,14 +135,18 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 		}
 		logger.V(3).Info("Using comparison time", "time", comparisonTime)
 
-		if comparisonTime.Add(timeoutForMachineToHaveNode).Before(now) {
+		if timeoutForMachineToHaveNode != time.Duration(0) && comparisonTime.Add(timeoutForMachineToHaveNode).Before(now) {
 			conditions.MarkFalse(t.Machine, clusterv1.MachineHealthCheckSuccededCondition, clusterv1.NodeStartupTimeoutReason, clusterv1.ConditionSeverityWarning, "Node failed to report startup in %s", timeoutForMachineToHaveNode.String())
 			logger.V(3).Info("Target is unhealthy: machine has no node", "duration", timeoutForMachineToHaveNode.String())
 			return true, time.Duration(0)
 		}
 
 		durationUnhealthy := now.Sub(comparisonTime)
 		nextCheck := timeoutForMachineToHaveNode - durationUnhealthy + time.Second
+		if timeoutForMachineToHaveNode == time.Duration(0) {
+			// Timeout is disabled, check again in 1 minute
+			nextCheck = time.Minute
```

Comment:
why 1 minute, shouldn't we just 0 here so this does not force a requeue and the MHC only react to updates in this machine/future associated node?

Reply:
This is a very good question and I can't quite remember why I did it, I initially wrote this implementation for a fork a few weeks back.

I think I was thinking about the general MHC status for healthy and unhealthy machine counts etc, we would want to make sure we update to reflect when the machine comes online. However, since we are watching the remote Nodes anyway, we will get an event to requeue if/when it comes online, this should return so that it doesn't requeue here. Good catch

Reply:
Agree with @enxebre here, I think we shouldn't requeue at all and keep the existing behavior of reacting to object updates only